### PR TITLE
Add inspect-physical-device command

### PR DESCRIPTION
**Pull request by shinnya**
See also https://github.com/frugalos/libfrugalos/pull/2.

# Message flow

frugalos/main(local)
-> frugalos/daemon(local)
-> frugalos/rpc_server
-> frugalos/service
-> frugalos/daemon(local)

# How to use

```console
$ frugalos inspect-physical-device --device dev1 --rpc-addr 127.0.0.1:3101
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -47,3 +47,6 @@ url = "1"
 
 [workspace]
 members = ["frugalos_config", "frugalos_mds", "frugalos_raft", "frugalos_segment"]
+
+[patch.crates-io]
+libfrugalos = { path = "../libfrugalos" }
```
Comment (shinnya) [inline Cargo.toml]: TODO delete this line

File: src/service.rs [hunk 2]
```diff
@@ -98,6 +101,38 @@ where
...
+                } else {
+                    None
+                }
+            }).ok_or_else(|| ErrorKind::InvalidInput.into())
+            .into_future();
+
+        future.and_then(|handle| {
+            handle
+                .request()
+                .list()
```
Comment (yuezato) [inline src/service.rs]: [For the sake of understanding]
Here we issue the `list` command to a device that has the given device-id passed as the command-line argument.
This line is one of the core parts of the present PR; right?

File: src/daemon.rs [hunk 3]
```diff
@@ -247,6 +264,19 @@ impl Future for DaemonRunner {
         while let Async::Ready(Some(command)) = self.command_rx.poll().expect("Never fails") {
             self.handle_command(command);
         }
+
+        // No sleep because this task is mainly for debugging.
```
Comment (yuezato) [inline src/daemon.rs]: retry

File: src/daemon.rs [hunk 4]
```diff
@@ -208,12 +214,15 @@ impl FrugalosDaemon {
...
 struct DaemonRunner {
+    logger: Logger,
     service: service::Service<ThreadPoolExecutorHandle>,
     http_server: HttpServer,
     rpc_server: fibers_rpc::server::Server<ThreadPoolExecutorHandle>,
     rpc_service: fibers_rpc::client::ClientService,
     command_rx: mpsc::Receiver<DaemonCommand>,
     stop_notifications: Vec<oneshot::Monitored<(), Error>>,
+    /// A sequence of tasks which inspect a physical device.
+    inspect_device_tasks: Vec<Box<Future<Item = (), Error = Error> + Send + 'static>>,
```
Comment (shinnya) [inline src/daemon.rs]: TODO: rename this variable and use it between other tasks.

File: src/daemon.rs (hunk 3 above)
Comment (shinnya) [reply src/daemon.rs]: fixed

File: src/daemon.rs (hunk 4 above)
Comment (shinnya) [reply src/daemon.rs]: I renamed this to `unimportant_tasks`.

File: src/service.rs (hunk 2 above)
Comment (shinnya) [reply src/service.rs]: Yes.

**Comment by shinnya** [comment]
Sorry reviewers. I think using kanils satisfies the need of this PR and close it.

**Pull request closed without merging**